src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala

```diff
@@ -114,7 +116,15 @@ object ErgoLikeTransactionSerializer extends SigmaSerializer[ErgoLikeTransaction
     val tokenIds = tx.outputCandidates.toColl
       .flatMap(box => box.additionalTokens.map(t => t._1))
 
-    val distinctTokenIds = tokenIds.distinct
+    val builder = mutable.ArrayBuilder.make[TokenId]()
+    val seen = mutable.Set[TokenId]()
+    tokenIds.foreach { ti =>
+      if (!seen.exists(v => util.Arrays.equals(v, ti))) {
+        builder += ti
+        seen.add(ti)
+      }
+    }
+    val distinctTokenIds = builder.result().toColl
```

Comment:
@greenhat We can move this logic to Coll library.
1) val res = tokenIds.map(id => Colls.fromArray(id)).distinct, this will give us Coll[Coll[Byte]]
2) res.map(_.toArray)

Reply:
Nice! Fixed.